R/bs-theme-preset.R

```diff
@@ -66,11 +61,34 @@ bs_preset_bundle.default <- function(preset, ...) {
 
 #' @export
 bs_preset_bundle.bs_preset <- function(preset, ...) {
-  switch(
-    preset$type %||% "",
-    builtin = builtin_bundle(preset$name, version = preset$version),
-    bootswatch = bootswatch_bundle(preset$name, version = preset$version),
-    NextMethod()
+  NextMethod()
+}
+
+#' @export
+bs_preset_bundle.bs_preset_builtin <- function(preset, ...) {
+  stopifnot(inherits(preset, "bs_preset"))
+  builtin_bundle(preset$name, version = preset$version)
+}
+
+#' @export
+bs_preset_bundle.bs_preset_bootswatch <- function(preset, ...) {
+  stopifnot(inherits(preset, "bs_preset"))
+  bootswatch_bundle(preset$name, version = preset$version)
+}
+
+theme_preset_info <- function(theme) {
+  if (!is_bs_theme(theme)) return(NULL)
+
+  theme_vars <- c("bslib-preset-type", "bslib-preset-name", "bootstrap-version")
+  info <- bs_get_variables(theme, theme_vars)
+
+  name <- if (!is.na(info[["bslib-preset-name"]])) info[["bslib-preset-name"]]
+  type <- if (!is.na(info[["bslib-preset-type"]])) info[["bslib-preset-type"]]
+
+  new_bs_preset(
+    name = name %||% "bootstrap",
```

Comment:
It kind of feels like we should define both `$bslib-preset-type` and `$bslib-preset-name` in `bs_theme_init()`. Not only to reduce the potential amount of places where we provide a default value, but also to make sure those variables are always defined in the Sass (so we can write Sass code that uses those variables and not worry about them being undefined)

Reply:
That's a good point and I'll have to think about it. The nice thing about the current implementation is that it's easy and to cleanly swap out preset layers, including changing those values